clients/egui/src/account/full_doc_search.rs

```diff
@@ -0,0 +1,295 @@
+
+            if output.response.changed() && !self.query.is_empty() {
+                self.results = vec![]; // return Some(*id.unwrap());
+
+                *self.is_searching.write().unwrap() = true;
+                self.send_search_results();
+            }
+
+            if self.is_searching.read().unwrap().eq(&true) {
+                ui.add_space(20.0);
+                ui.spinner();
+            }
+
+            if self.query.is_empty() {
+                self.results = vec![];
+            }
+
+            if !self.results.is_empty() {
+                return egui::ScrollArea::vertical()
+                    .show(ui, |ui| self.show_results(ui, core))
+                    .inner;
+            };
+
+            None
+        })
+        .inner
+    }
+
+    fn send_search_results(&self) {
+        let search_tx = self.search_channel.search_tx.clone();
+        let results_tx = self.results_tx.clone();
+        let results_rx = self.search_channel.results_rx.clone();
+        let query = self.query.clone();
+        let is_searching = self.is_searching.clone();
+
+        thread::spawn(move || {
+            search_tx
+                .send(SearchRequest::Search { input: query })
+                .unwrap();
+
+            while let Ok(sr) = results_rx.recv() {
+                results_tx.send(sr).unwrap();
+                match results_rx.try_recv() {
+                    Err(e) if e.is_empty() => {
+                        *is_searching.write().unwrap() = false;
+                    }
+                    _ => (),
+                }
+            }
+        });
+    }
+
+    pub fn show_results(&mut self, ui: &mut egui::Ui, core: &lb::Core) -> Option<&lb::Uuid> {
+        ui.add_space(20.0);
+
+        for (_, sr) in self.results.iter().enumerate() {
+            let sr_res = ui.vertical(|ui| {
+                match sr {
+                    Error(err) => {
+                        ui.horizontal(|ui| {
+                            ui.add_space(self.x_margin);
+                            ui.label(
+                                egui::RichText::new(err.msg.to_owned())
+                                    .color(ui.visuals().extreme_bg_color),
+                            );
+                        });
+                    }
+                    FileNameMatch { id, path, matched_indices: _, score: _ } => {
+                        let file = &core.get_file_by_id(*id).unwrap();
+                        Self::show_file(ui, file, path, self.x_margin);
+                    }
+
+                    FileContentMatches { id, path, content_matches } => {
+                        let file = &core.get_file_by_id(*id).unwrap();
+                        Self::show_file(ui, file, path, self.x_margin);
+                        ui.horizontal(|ui| {
+                            ui.add_space(15.0);
+                            ui.horizontal_wrapped(|ui| {
+                                self.show_content_match(ui, &content_matches[0], 15.0);
```

Comment:
Only the first content match item for a file will be shown this way. Ideally, we want to be able to see each content match for a file as their own search result, with their own score.

Reply:
how about showing the content match with the highest score and ignoring the other content matches (from the same file).
 
The goal of "file content search" is not to locate and jump between text content in a certain file (like in a simple `<C, f>`), but to locate _the file path_. we display the body content match just for context info. 

If multiple entries are added per file, that might crowd out the file path search results

Reply:
I don't think showing just the highest score, at least with our current implementation of "file content search," is necessarily the best option. I return multiple content search results so that users can peer through them and see whether a file contains the specific body of text they want. This is a step more important than just functioning as a context info for a _file path_.

Although I do certainly see what you mean, and this very well can crowd up search results with multiple matches from the same file(s). I also have thoughts on how search can be improved, and we can definitely come back to this, but I feel for the sake of consistency with other platforms, its best to separate the results.